internal/objects/contour/contour.go

```diff
@@ -0,0 +1,123 @@
+// Copyright Project Contour Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package contour
+
+import (
+	"context"
+	"fmt"
+
+	operatorv1alpha1 "github.com/projectcontour/contour-operator/api/v1alpha1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// New makes a Contour object using the provided ns/name for the object's
+// namespace/name, pubType for the network publishing type of Envoy, and
+// Envoy container ports 8080/8443.
+func New(name, ns, specNs string, remove bool, pubType operatorv1alpha1.NetworkPublishingType) *operatorv1alpha1.Contour {
+	return &operatorv1alpha1.Contour{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: ns,
+			Name:      name,
+		},
+		Spec: operatorv1alpha1.ContourSpec{
+			Namespace: operatorv1alpha1.NamespaceSpec{
+				Name:             specNs,
+				RemoveOnDeletion: remove,
+			},
+			NetworkPublishing: operatorv1alpha1.NetworkPublishing{
+				Envoy: operatorv1alpha1.EnvoyNetworkPublishing{
+					Type: pubType,
+					ContainerPorts: []operatorv1alpha1.ContainerPort{
+						{
+							Name:       "http",
+							PortNumber: int32(8080),
+						},
+						{
+							Name:       "https",
+							PortNumber: int32(8443),
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+// OwnerLabelsExist returns true if obj contains Contour owner labels.
+func OwnerLabelsExist(obj metav1.Object, contour *operatorv1alpha1.Contour) bool {
```

Comment:
Refactor this into instance methods to make it cleaner maybe? 

e.g.
```
func (c *operatorv1alpha1.Contour)OwnerLabelsExist(obj metav1.Object) bool {}
```

Reply:
I cannot create an instance method on a non-local type. I need to wrap the external type and do the same for other pkgs that call OwnerLabelsExist(). Thoughts?

Reply:
I think we should either just do this, or put these functions in the exported API. If anyone else is doing things with Contours, they'll be handy anyway.

Reply:
Contour put a bunch of helpers like that in the API (https://github.com/projectcontour/contour/blob/main/apis/projectcontour/v1/helpers.go). My thought is just that it's cleaner to say `contour.OwnerLabelsExist(obj)` vs passing around the `*operatorv1alpha1.Contour` each time. 

Could be a future refactor as well if needed.

Reply:
Probably better the way it is than public in the api.